api/protogen/protogen.go

```diff
@@ -270,12 +273,55 @@ func terraformAttributeTypeToProtoType(attrName string, attrType attr.Type) (rep
 		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
 	case types.SetType:
 		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
-	// TODO: Add support for nested objects.
+	case types.ObjectType:
+		return terraformObjectAttributeTypeToProtoType(attrName, v)
+
 	default:
 		return false, "", nil, fmt.Errorf("unsupported Terraform type: %s", attrType.String())
 	}
 }
 
+// terraformObjectAttributeTypeToProtoType converts a Terraform object attribute to a Protocol Buffer message type.
+func terraformObjectAttributeTypeToProtoType(attrName string, obj types.ObjectType) (repeated bool, protoType string, messages *message, err error) {
+	messageName := schema.ProtoMessageName(attrName)
+	newMessage := &message{
+		Name:   messageName,
+		Fields: []field{},
+	}
+
+	for name, attrType := range obj.AttrTypes {
+		isRepeated, t, msg, err := terraformAttributeTypeToProtoType(name, attrType)
+		if err != nil {
+			return false, "", nil, fmt.Errorf("failed to parse field %s in object %s: %w", name, attrName, err)
+		}
+
+		newMessage.Fields = append(newMessage.Fields, field{
+			Repeated: isRepeated,
+			Type:     t,
+			Name:     name,
+			Tag:      len(newMessage.Fields) + 1,
```

Comment:
You must use the `apiSpecTagger` to assign tags, and to preserve tags when new fields are added into fields, to ensure backward compatibility. This PR must not break this feature.

Reply:
fixed